plugin/storage/es/spanstore/writer.go

```diff
@@ -104,20 +106,28 @@ func NewSpanWriter(p SpanWriterParams) *SpanWriter {
 		spanIndexPrefix:    p.IndexPrefix + spanIndex,
 		serviceIndexPrefix: p.IndexPrefix + serviceIndex,
 		spanConverter:      dbmodel.NewFromDomain(p.AllTagsAsFields, p.TagKeysAsFields, p.TagDotReplacement),
+		archive:            p.Archive,
 	}
 }
 
 // WriteSpan writes a span and its corresponding service:operation in ElasticSearch
 func (s *SpanWriter) WriteSpan(span *model.Span) error {
-	spanIndexName := indexWithDate(s.spanIndexPrefix, span.StartTime)
-	serviceIndexName := indexWithDate(s.serviceIndexPrefix, span.StartTime)
+	var spanIndexName string
+	if s.archive {
+		spanIndexName = archiveIndex(s.spanIndexPrefix)
+	} else {
+		spanIndexName = indexWithDate(s.spanIndexPrefix, span.StartTime)
```

Comment:
seems like if we merged with PR with index name pattern, we would not have to make these splits between archive/non-archive.

Reply:
yes this is a bit ugly. We can set the func in the constructor.